diff -rupwN -x .svn ../dnl3/ffmpeg/libavcodec/h264dspenc.c ffmpeg/libavcodec/h264dspenc.c
--- ../dnl3/ffmpeg/libavcodec/h264dspenc.c	2010-03-25 11:58:09.000000000 -0400
+++ ffmpeg/libavcodec/h264dspenc.c	2010-03-25 11:46:17.000000000 -0400
@@ -73,6 +73,6 @@ static void h264_dct_c(DCTELEM block[4][
 
 av_cold void ff_h264dspenc_init(DSPContext* c, AVCodecContext *avctx)
 {
-    c->h264_dct = h264_dct_c;
+//    c->h264_dct = h264_dct_c;
 }
 
diff -rupwN -x .svn ../dnl3/ffmpeg/libavcodec/h264enc.c ffmpeg/libavcodec/h264enc.c
--- ../dnl3/ffmpeg/libavcodec/h264enc.c	2010-03-25 11:58:09.000000000 -0400
+++ ffmpeg/libavcodec/h264enc.c	2010-03-25 11:47:54.000000000 -0400
@@ -184,7 +184,7 @@ static inline int quantize_c(DCTELEM *bl
                              int intra, int separate_dc)
 {
     int i;
-    const int * const quant_3Btable = quant_coeff[qscale];
+    const int * const quant_table = quant_coeff[qscale];
     const int bias = intra ? (1 << QUANT_SHIFT) / 3 : (1 << QUANT_SHIFT) / 6;
     const unsigned int threshold1 = (1 << QUANT_SHIFT) - bias - 1;
     const unsigned int threshold2 = (threshold1 << 1);
diff -rupwN -x .svn ../dnl3/ffmpeg/libavcodec/mpegaudio_tablegen.c ffmpeg/libavcodec/mpegaudio_tablegen.c
--- ../dnl3/ffmpeg/libavcodec/mpegaudio_tablegen.c	2010-03-25 11:58:09.000000000 -0400
+++ ffmpeg/libavcodec/mpegaudio_tablegen.c	2010-03-25 11:53:41.000000000 -0400
@@ -22,6 +22,7 @@
 
 #include <stdlib.h>
 #define CONFIG_HARDCODED_TABLES 0
+#include "mpegaudio.h"
 #include "mpegaudio_tablegen.h"
 #include "tableprint.h"
 
diff -rupwN -x .svn ../dnl3/ffmpeg/libavutil/avconfig.h ffmpeg/libavutil/avconfig.h
--- ../dnl3/ffmpeg/libavutil/avconfig.h	1969-12-31 19:00:00.000000000 -0500
+++ ffmpeg/libavutil/avconfig.h	2010-03-25 09:51:26.000000000 -0400
@@ -0,0 +1,4 @@
+#ifndef AVUTIL_AVCONFIG_H
+#define AVUTIL_AVCONFIG_H
+#define AV_HAVE_BIGENDIAN 0
+#endif /* AVUTIL_AVCONFIG_H */
