diff -rupwbBEN --strip-trailing-cr -x .svn ../dnl3/irrlicht/source/Irrlicht/CIrrDeviceConsole.cpp irrlicht/source/Irrlicht/CIrrDeviceConsole.cpp
--- ../dnl3/irrlicht/source/Irrlicht/CIrrDeviceConsole.cpp	2011-02-04 10:06:25.000000000 -0500
+++ irrlicht/source/Irrlicht/CIrrDeviceConsole.cpp	2011-02-03 10:41:13.000000000 -0500
@@ -75,6 +75,7 @@ CIrrDeviceConsole::CIrrDeviceConsole(con
 	WindowsSTDOut = GetStdHandle(STD_OUTPUT_HANDLE);
 	PCOORD Dimensions = 0;
 
+#if _WIN32_WINNT > 0x0501	
 	if (CreationParams.Fullscreen)
 	{
 		if (SetConsoleDisplayMode(WindowsSTDOut, CONSOLE_FULLSCREEN_MODE, Dimensions))
@@ -84,6 +85,7 @@ CIrrDeviceConsole::CIrrDeviceConsole(con
 		}
 	}
 	else
+#endif
 	{
 		COORD ConsoleSize;
 		ConsoleSize.X = CreationParams.WindowSize.Width;
diff -rupwbBEN --strip-trailing-cr -x .svn ../dnl3/irrlicht/source/Irrlicht/CIrrDeviceWin32.h irrlicht/source/Irrlicht/CIrrDeviceWin32.h
--- ../dnl3/irrlicht/source/Irrlicht/CIrrDeviceWin32.h	2011-02-04 10:06:25.000000000 -0500
+++ irrlicht/source/Irrlicht/CIrrDeviceWin32.h	2011-02-04 09:47:55.000000000 -0500
@@ -16,7 +16,7 @@
 #if !defined(_IRR_XBOX_PLATFORM_)
 	#include <windows.h>
 	#include <mmsystem.h> // For JOYCAPS
-	#include <Windowsx.h>
+	#include <windowsx.h>
 #endif
 
 namespace irr
diff -rupwbBEN --strip-trailing-cr -x .svn ../dnl3/irrlicht/source/Irrlicht/Irrlicht.cpp irrlicht/source/Irrlicht/Irrlicht.cpp
--- ../dnl3/irrlicht/source/Irrlicht/Irrlicht.cpp	2011-02-04 10:06:25.000000000 -0500
+++ irrlicht/source/Irrlicht/Irrlicht.cpp	2011-02-03 17:54:02.000000000 -0500
@@ -125,7 +125,7 @@ namespace video
 
 } // end namespace irr
 
-
+/*
 #if defined(_IRR_WINDOWS_API_)
 
 BOOL APIENTRY DllMain( HANDLE hModule,
@@ -150,4 +150,5 @@ BOOL APIENTRY DllMain( HANDLE hModule,
 }
 
 #endif // defined(_IRR_WINDOWS_)
+*/
 
