diff -rupwN -x .svn ../dnl3/irrlicht/source/Irrlicht/CImageLoaderPNG.cpp irrlicht/source/Irrlicht/CImageLoaderPNG.cpp
--- ../dnl3/irrlicht/source/Irrlicht/CImageLoaderPNG.cpp	2010-03-25 12:17:20.000000000 -0400
+++ irrlicht/source/Irrlicht/CImageLoaderPNG.cpp	2010-03-25 12:13:07.000000000 -0400
@@ -162,7 +162,7 @@ IImage* CImageLoaderPng::loadImage(io::I
 	if (BitDepth < 8)
 	{
 		if (ColorType==PNG_COLOR_TYPE_GRAY || ColorType==PNG_COLOR_TYPE_GRAY_ALPHA)
-			png_set_gray_1_2_4_to_8(png_ptr);
+			png_set_expand_gray_1_2_4_to_8(png_ptr);
 		else
 			png_set_packing(png_ptr);
 	}
diff -rupwN -x .svn ../dnl3/irrlicht/source/Irrlicht/Irrlicht.cpp irrlicht/source/Irrlicht/Irrlicht.cpp
--- ../dnl3/irrlicht/source/Irrlicht/Irrlicht.cpp	2010-03-25 12:17:20.000000000 -0400
+++ irrlicht/source/Irrlicht/Irrlicht.cpp	2010-03-25 12:13:36.000000000 -0400
@@ -126,7 +126,7 @@ namespace video
 } // end namespace irr
 
 
-#if defined(_IRR_WINDOWS_API_)
+#if defined(_IRR_WINDOWS_API_) && !defined( _IRR_STATIC_LIB_ )
 
 BOOL APIENTRY DllMain( HANDLE hModule,
                        DWORD  ul_reason_for_call,
